bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/MirroringTable.java

```diff
@@ -0,0 +1,1052 @@
+      Log.trace(
+          "[%s] checkAndPut(row=%s, family=%s, qualifier=%s, compareOp=%s, value=%s, put=%s)",
+          this.getName(), row, family, qualifier, compareOp, value, put);
+      RowMutations mutations = new RowMutations(row);
+      mutations.add(put);
+      return this.checkAndMutateWithSpan(row, family, qualifier, compareOp, value, mutations);
+    }
+  }
+
+  @Override
+  public void delete(final Delete delete) throws IOException {
+    try (Scope scope = this.mirroringTracer.spanFactory.operationScope(HBaseOperation.DELETE)) {
+      Log.trace("[%s] delete(delete=%s)", this.getName(), delete);
+      this.batchSingleWriteOperation(delete);
+    }
+  }
+
+  @Override
+  public void delete(List<Delete> deletes) throws IOException {
+    try (Scope scope =
+        this.mirroringTracer.spanFactory.operationScope(HBaseOperation.DELETE_LIST)) {
+      Log.trace("[%s] delete(deletes=%s)", this.getName(), deletes);
+      // Delete should remove successfully deleted rows from input list.
+      Object[] results = new Object[deletes.size()];
+      try {
+        this.batchWithSpan(deletes, results);
+      } catch (InterruptedException e) {
+        IOException e2 = new InterruptedIOException();
+        e2.initCause(e);
+        throw e2;
+      } finally {
+        final FailedSuccessfulSplit<Delete> failedSuccessfulSplit =
+            new FailedSuccessfulSplit<>(deletes, results, resultIsFaultyPredicate);
+
+        deletes.clear();
+        deletes.addAll(failedSuccessfulSplit.failedOperations);
+      }
+    }
+  }
+
+  @Override
+  public boolean checkAndDelete(
+      byte[] row, byte[] family, byte[] qualifier, byte[] value, Delete delete) throws IOException {
+    Log.trace(
+        "[%s] checkAndDelete(row=%s, family=%s, qualifier=%s, value=%s, delete=%s)",
+        this.getName(), row, family, qualifier, value, delete);
+    return this.checkAndDelete(row, family, qualifier, CompareOp.EQUAL, value, delete);
+  }
+
+  @Override
+  public boolean checkAndDelete(
+      byte[] row, byte[] family, byte[] qualifier, CompareOp compareOp, byte[] value, Delete delete)
+      throws IOException {
+    try (Scope scope =
+        this.mirroringTracer.spanFactory.operationScope(HBaseOperation.CHECK_AND_DELETE)) {
+      Log.trace(
+          "[%s] checkAndDelete(row=%s, family=%s, qualifier=%s, compareOp=%s, value=%s, delete=%s)",
+          this.getName(), row, family, qualifier, compareOp, value, delete);
+      RowMutations mutations = new RowMutations(row);
+      mutations.add(delete);
+      return this.checkAndMutateWithSpan(row, family, qualifier, compareOp, value, mutations);
+    }
+  }
+
+  @Override
+  public void mutateRow(final RowMutations rowMutations) throws IOException {
+    try (Scope scope = this.mirroringTracer.spanFactory.operationScope(HBaseOperation.MUTATE_ROW)) {
+      Log.trace("[%s] mutateRow(rowMutations=%s)", this.getName(), rowMutations);
+      batchSingleWriteOperation(rowMutations);
```

Comment:
I dont think we should generalize point writes to batch writes. Awhile back Solomon noticed that there was a performance penality in bigtable (not sure if its still the case), but also it will misleading in metrics

Reply:
It it hard to modify it now, we will leave it as is.

Reply:
will add a TODO. No  budget to fix now.